signedTransaction.go

```diff
@@ -75,6 +75,12 @@ func (txn *SignedTransaction) MarshalBCS(ser *bcs.Serializer) {
 	txn.Authenticator.MarshalBCS(ser)
 }
 func (txn *SignedTransaction) UnmarshalBCS(des *bcs.Deserializer) {
+	if txn.Transaction == nil {
+		txn.Transaction = &RawTransaction{}
+	}
+	if txn.Authenticator == nil {
+		txn.Authenticator = &TransactionAuthenticator{}
+	}
```

Comment:
Could probably just initialize them without checking they're null, since a deserialize will knock out the previous value.